contracts/utils/Base64.sol

```diff
@@ -87,4 +89,68 @@ library Base64 {
 
         return result;
     }
+
+    /**
+     * @dev Converts a `bytes` to its Bytes64Url `string` representation.
+     */
+    function encodeUrl(bytes memory data) internal pure returns (string memory) {
+        /**
+         * Inspired by Brecht Devos (Brechtpd) implementation - MIT licence
+         * https://github.com/Brechtpd/base64/blob/e78d9fd951e7b0977ddca77d92dc85183770daf4/base64.sol
+         */
+        if (data.length == 0) return "";
+
+        // Loads the table into memory
+        string memory table = _TABLE_URL;
+
+        // Encoding takes 3 bytes chunks of binary data from `bytes` data parameter
+        // and split into 4 numbers of 6 bits.
+        // Rounding here is different from the one performed in {encode} because we don't want padding
```

Comment:
I see why this rounding is different. But I'm thinking the wording of the `encode` function should be here instead:

```suggestion
        // The final Base64URL length should be `bytes` data length multiplied by 4/3 rounded up
        // as opposed to Base64 rounding where padding is required to fill the last chunk.
        // - `4 *`              -> 4 characters for each chunk
        // - `data.length + 2`  -> Round up
        // - `/ 3`              -> Number of 3-bytes chunks
```

And we should add the following to the `encode` function instead:

```
// The final Base64 length should be `bytes` data length divided by 3 rounded up and then multiplied by 4
// so that it leaves room for padding the last chunk
// - `data.length + 2`  -> Round up
// - `/ 3`              -> Number of 3-bytes chunks
// - `4 *`              -> 4 characters for each chunk
```